### DISCUSSION TITLE
Checking parameters from 2 different data sets

### DISCUSSION THREAD

**Discussion by Miurath** [eclipse-birt/birt#1008] [Q&A]
I am trying to check a parameter from different datasets A or B and its result should be saved in a variable (or alternatively just be used in a visibility-check).

There is always a result in A or B or (A and B). A and B always have the parameter I want to read filled.

Since I cannot bind 2 datasets to a single element I would need to stack elements with different bindings together to make this work.

Is there a more elegant way to do it?

**Comment by hvbtup** [comment]
Stacking the elements in the layout as you mentioned is one option:

    List A  (uses DS A, sets variable)
      detail:
        List B (useds DS B, sets variable)
          detail:
            List/Table C (DS needs the variable)

Another option without stacking is to create two lists A and B for the two data data sets _above_ the list/table item C where you need the parameter.
Hide A and B using the visibility expression (assuming you just want them for checking the parameter) and set the variable in the onCreate script of A and B accordingly.

List A (uses DS A, hidden, sets variable in onCreate)
List B (uses DS A, hidden, sets variable in onCreate)
List/Table C (whose DS needs needs the variable)

Then you can use that variable as an input parameter for C.

If you are using JDBC datasets, you can benefit from BIRT's caching, e.g. if you also need to show the results of A and/or B, the queries wil only be executed once in the DB.

**Comment by TheLaserNinja** [comment]
I need a little more context. Is the values coming from data-sets? If not, why couldn't you just use report parameters? Report parameters are global as opposed to data set parameters which are only scoped within the data set. Or, in the on-fetch event, set a global variable? Can you explain the actual workflow?
